functional-tests.sh

```diff
@@ -530,6 +530,18 @@ function test_config_host_add_error()
     log_success "$start_time" "${FUNCNAME[0]}"
 }
 
+function test_put_object_with_storage_class()
+{
+    show "${FUNCNAME[0]}"
+
+    start_time=$(get_time)
+    object_name="mc-test-object-$RANDOM"
+    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp --storage-class REDUCED_REDUNDANCY "${FILE_1_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rm "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
+
+    log_success "$start_time" "${FUNCNAME[0]}"
```

Comment:
can you add a negative test with invalid storage class as well?

Reply:
done